sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -308,6 +308,11 @@ public Iterable<String> keys(Function<String, String> headerGetter) {
         @Override
         @SuppressWarnings("deprecation")
         public String get(Function<String, String> headerGetter, String headerName) {
+            if (headerGetter == null) {
+                // headerGetter is annotated with Nullable, guard.
```

Comment:
Where is `headerGetter` annotated as `Nullable`? I could not find that in the class. I don't think it's inherently wrong to add extra checks, I just wonder changing from throwing an NPE in the following lines would too meaningful of a behavioral breaking change.

Reply:
In all honesty I would just let things fail if we get a null `headerGetter`.

Reply:
It's not annotated here as `Nullable` but in the super type this is implements/overidding